pkg/server/server.go

```diff
@@ -260,13 +308,24 @@ func (s *Server) StreamedListObjects(req *openfgav1.StreamedListObjectsRequest,
 		return err
 	}
 
+	var checkCache *ccache.Cache[*graph.ResolveCheckResponse]
+
+	if slices.Contains(s.experimentals, ExperimentalCacheResolveCheck) {
+		if s.checkQueryCacheEnabled {
+			checkCache = s.checkCache
+			// otherwise when we pass nil to the list object query, the list query will not use cache
+		}
+	}
```

Comment:
```suggestion
	if slices.Contains(s.experimentals, ExperimentalCacheResolveCheck) && s.checkQueryCacheEnabled {
		checkCache = s.checkCache
	}
```

Reply:
I purposely split them as the experimental flag will be removed